apps/wallet-mobile/src/yoroi-wallets/types/other.ts

```diff
@@ -417,7 +417,7 @@ export type TransactionInfo = {
   tokens: Record<string, Token>
   blockNumber: number
   memo: null | string
-  metadata?: TxMetadataInfo
+  metadata: TxMetadataInfo | undefined
```

Comment:
isn't it `| null` just checking

Reply:
Hi, only for `TxMetadataInfo | undefined` the TS is not complaining - to be null we need to manually add it when returning from `processTxHistoryData` 🤔

Reply:
What's the difference between 
```ts
metadata?: TxMetadataInfo
```
and
```ts
metadata: TxMetadataInfo | undefined
```
?

They should be equivalent.

Reply:
It's kind of, the first states "metadata" is optional in the object, the second "metadata" is required AS PROPERTY in the object, yet it can be `undefined`

Reply:
it doesn't change the validation, yet it changes the data representation, cuz one expects the property to be present even if it is empty

Reply:
k, @SorinC6 `tx.metadata?.reduce` can make it `undefined` missed the`?`